src/gui/filedetails/sharemodel.cpp

```diff
@@ -0,0 +1,960 @@
+        return;
+    }
+
+    const auto sharePersistentModelIndex = _shareIdIndexHash.value(shareId);
+    const auto shareModelIndex = index(sharePersistentModelIndex.row());
+    Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { EditingAllowedRole });
+}
+
+void ShareModel::slotSharePasswordSet(const QString &shareId)
+{
+    if (shareId.isEmpty() || !_shareIdIndexHash.contains(shareId)) {
+        return;
+    }
+
+    const auto sharePersistentModelIndex = _shareIdIndexHash.value(shareId);
+    const auto shareModelIndex = index(sharePersistentModelIndex.row());
+    Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { PasswordProtectEnabledRole, PasswordRole });
+}
+
+void ShareModel::slotShareNoteSet(const QString &shareId)
+{
+    if (shareId.isEmpty() || !_shareIdIndexHash.contains(shareId)) {
+        return;
+    }
+
+    const auto sharePersistentModelIndex = _shareIdIndexHash.value(shareId);
+    const auto shareModelIndex = index(sharePersistentModelIndex.row());
+    Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { NoteEnabledRole, NoteRole });
+}
+
+void ShareModel::slotShareNameSet(const QString &shareId)
+{
+    if (shareId.isEmpty() || !_shareIdIndexHash.contains(shareId)) {
+        return;
+    }
+
+    const auto sharePersistentModelIndex = _shareIdIndexHash.value(shareId);
+    const auto shareModelIndex = index(sharePersistentModelIndex.row());
+    Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { LinkShareNameRole });
+}
+
+void ShareModel::slotShareLabelSet(const QString &shareId)
+{
+    if (shareId.isEmpty() || !_shareIdIndexHash.contains(shareId)) {
+        return;
+    }
+
+    const auto sharePersistentModelIndex = _shareIdIndexHash.value(shareId);
+    const auto shareModelIndex = index(sharePersistentModelIndex.row());
+    Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { Qt::DisplayRole, LinkShareLabelRole });
+}
+
+void ShareModel::slotShareExpireDateSet(const QString &shareId)
+{
+    if (shareId.isEmpty() || !_shareIdIndexHash.contains(shareId)) {
+        return;
+    }
+
+    const auto sharePersistentModelIndex = _shareIdIndexHash.value(shareId);
+    const auto shareModelIndex = index(sharePersistentModelIndex.row());
+    Q_EMIT dataChanged(shareModelIndex, shareModelIndex, { ExpireDateEnabledRole, ExpireDateRole });
+}
+
+// ----------------------- Shares modification slots ----------------------- //
+
+void ShareModel::toggleShareAllowEditing(const SharePtr &share, const bool enable) const
+{
+    if (share.isNull()) {
+        return;
+    }
+
+    auto permissions = share->getPermissions();
+    enable ? permissions |= SharePermissionUpdate : permissions &= ~SharePermissionUpdate;
+
+    share->setPermissions(permissions);
+}
+
+void ShareModel::toggleShareAllowEditingFromQml(const QVariant &share, const bool enable) const
+{
+    const auto ptr = share.value<SharePtr>();
+    toggleShareAllowEditing(ptr, enable);
+}
```

Comment:
We can try but the issue is that exposing SharePermissions to QML results in some very inelegant wrapper object being needed, with more processing in the UI files to set and check flags. It also doesn't map well to the UI components (i.e. a simple slot called by clicking on the button)

Additionally since at the moment we are only using one slot in the UI we don't really need a "generic" slot

I am happy to still make this change is it is disagreeable